projects/plugins/boost/app/modules/optimizations/lcp/class-lcp.php

```diff
@@ -377,6 +378,55 @@ private function optimize_image( $tag ) {
 			$tag = preg_replace( '/<img\s/i', '<img loading="eager" ', $tag );
 		}
 
+		$tag = $this->optimize_image_tag_srcset( $tag );
+
+		return $tag;
+	}
+
+	/**
+	 * Optimize an image tag by adding srcset and sizes attributes.
+	 *
+	 * @param string $tag The original image tag.
+	 * @return string The optimized image tag.
+	 *
+	 * @since $$next-version$$
+	 */
+	private function optimize_image_tag_srcset( $tag ) {
+		if ( preg_match( '/src\s*=\s*["\']([^"\']+)["\']/i', $tag, $matches ) ) {
+			$image_url = $matches[1];
+
+			$image_sizes = array( 300, 600, 900, 1200, 1600 );
+			$sizes       = array();
+			foreach ( $image_sizes as $size ) {
+				$sizes[ $size ] = array( 'w' => $size );
+			}
+
+			// Generate srcset
+			$srcset = array();
+			foreach ( $sizes as $width => $args ) {
+				$srcset[] = Image_CDN_Core::cdn_url( $image_url, $args ) . ' ' . $width . 'w';
+			}
+
+			// Add srcset attribute
+			if ( ! preg_match( '/srcset\s*=\s*["\'][^"\']*["\']/i', $tag ) ) {
```

Comment:
There's very likely going to be a srcset already (which might not be the ideal for the img) so we would want to override that as well. Boost knows best what's the right srcset for the LCP.

Reply:
Good point. I'll do that with the sizes too as you said below.